source/Octopus.Client.Tests/Operations/RegisterKubernetesClusterOperationFixture.cs

```diff
@@ -101,5 +108,147 @@ public async Task ShouldCreateNewPollingKubernetesTentacle()
 
             machineResources.Should().ContainSingle().Which.Endpoint.Should().BeOfType<KubernetesTentacleEndpointResource>().Which.Should().BeEquivalentTo(new KubernetesTentacleEndpointResource(new PollingTentacleEndpointConfigurationResource("ABCDEF", "poll://ckyhfyfkcbmzjl8sfgch/")));
         }
+
+        [Test]
+        public async Task ShouldOnlyUpdateEndpointConnectionConfigurationOnExistingMachine()
```

Comment:
I ended up adding all these tests from the `RegisterMachineOperationFixture` just to ensure the functionality is maintained for `RegisterKubernetesClusterOperation`, even if someone wants to change `RegisterMachineOperation`.